Engine/Setup/BrokerageSetupHandler.cs

```diff
@@ -306,9 +309,43 @@ public bool Setup(IAlgorithm algorithm, IBrokerage brokerage, AlgorithmNodePacke
                         if (!algorithm.Portfolio.ContainsKey(holding.Symbol))
                         {
                             Log.Trace("BrokerageSetupHandler.Setup(): Adding unrequested security: " + holding.Symbol.ToString());
-                            // for items not directly requested set leverage to 1 and at the min resolution
-                            algorithm.AddSecurity(holding.Type, holding.Symbol.Value, minResolution.Value, null, true, 1.0m, false);
+
+                            var marketHoursDatabase = MarketHoursDatabase.FromDataFolder();
+                            var symbolPropertiesDatabase = SymbolPropertiesDatabase.FromDataFolder();
+
+                            // if we adding option, we have to add the option universe, so we take underlying symbol
+                            // Implemented solution is temporary. We will remove this code once it is the idea on how to refactor IAlgorithm is matured. 
+                            if (holding.Type == SecurityType.Option)
```

Comment:
Curiosity - What kinds of ideas do we have for refactoring `IAlgorithm` to better support this in a less fragile/highly coupled manner?